python/ray/autoscaler/command_runner.py

```diff
@@ -53,13 +53,14 @@ class CommandRunnerInterface:
 
     Command runner instances are returned by provider.get_command_runner()."""
 
-    def run(self,
+    def run(
+            self,
             cmd: str = None,
             timeout: int = 120,
             exit_on_fail: bool = False,
             port_forward: List[Tuple[int, int]] = None,
             with_output: bool = False,
-            **kwargs) -> str:
```

Comment:
My only problem is that by removing the `kwargs`, is that `run_env` and `ssh_options_override_ssh_key` only appear in some `run()` calls.

Reply:
Why is that a problem if they have default values?

Reply:
run_env is somewhat confusing, how come its default is `True`, but it is assigned str values?

Reply:
Oh, `run_env` should be defaulted to "auto". The concern is that `run` is called with `run_env="auto"` for **all `command_runners`**, not the ones where `run_env` is defined (`SSHCommandRunner` & `KubernetesCommandRunner`, etc.)

Reply:
I added run_env to all the command runners. I also removed unnecessary run_env arguments when used with commands runners that don't actually use them like SSHCommandRunner

Reply:
Sounds good to me!